### PR TITLE
Extended printf test to verify length sub-specifiers

### PR DESCRIPTION
**Pull request by shajder**
According to work plan for issue #1058

### CODE REVIEW

File: test_conformance/printf/util_printf.cpp [hunk 1]
```diff
@@ -1397,6 +1397,104 @@ testCase testCaseMixedFormat = { TYPE_MIXED_FORMAT_RANDOM,
...
+
+//=============================================================
+
+// length sub-specifier format
+
+//=============================================================
+
+std::vector<printDataGenParameters> printLenSpecGenParameters = {
+
+    { { "%hd" }, "32767" },
```
Comment (svenvh) [inline test_conformance/printf/util_printf.cpp:1409]: Do we want to test `%hd`, `%hx` and `%ho` with a negative value too?

Comment (shajder) [reply test_conformance/printf/util_printf.cpp:1409]: Good point - it makes sense, my only concern is if this should be added here along with sub-specifiers test or in actual `TYPE_INT`/`TYPE_OCTAL`/`TYPE_HEXADEC`. For some reason neither of those tests take into account negative values.

Comment (bashbaug) [reply test_conformance/printf/util_printf.cpp:1409]: I'd prefer to test positive and negative values here so all of the length modifier testing is in one place.

We probably should test negative numbers for the `int`, `hexadecimal`, and `octal` tests also, but we can do that in a separate PR.

**Comment by bashbaug** [comment]
Note, has a merge conflict that needs to be resolved.

File: test_conformance/printf/util_printf.cpp (hunk 1 above)
Comment (shajder) [reply test_conformance/printf/util_printf.cpp:1477]: Done! Soon I will prepare complementation PR for missing negative values in previous test cases

**Comment by shajder** [comment]
> Note, has a merge conflict that needs to be resolved.

Conflicts solved

**Review by lakshmih — APPROVED** [review]

**Review by svenvh — APPROVED** [review]

**Review by bashbaug — APPROVED** [review]

**Comment by bashbaug** [comment]
Merging as discussed in the October 8th teleconference.

**Pull request merged**